var/spack/repos/builtin/packages/timemory/package.py

```diff
@@ -22,18 +23,61 @@ class Timemory(CMakePackage):
     version('3.0.0', commit='b36b1673b2c6b7ff3126d8261bef0f8f176c7beb',
             submodules=True)
 
+    linux = False if platform == 'darwin' else True
+
+    variant('shared', default=True, description='Build shared libraries')
+    variant('static', default=False, description='Build static libraries')
```

Comment:
You'll prob want a conflict for `~shared~static`, as I assume that combination isn't possible.

Reply:
It is actually possible. It is a template library so it can actually function as header only although it's not really recommended bc w/o any extern template instantiations the compile times get pretty long when downstream applications use it.